app/handlers/sessions_callback.rb

```diff
@@ -102,24 +102,24 @@ def handle
         run(TransferAuthentications, authentication, current_user)
         status = (current_user.is_temp ? :new_user : :returning_user)
       else
-        outcome = run(CreateUserFromOmniauth, @auth_data)
+        outcome = run(CreateUserFromOmniauthData, @data)
         new_user = outcome.outputs[:user]
         run(TransferAuthentications, authentication, new_user)
-        sign_in(new_user)
+        sign_in!(new_user)
         status = :new_user
       end
 
     end
 
     if this_authentication_is_new
-      run(TransferOmniauthInformation, @auth_data, current_user)
+      run(AddEmailFromOmniauthData, @data, current_user)
```

Comment:
My first question was going to be "Can this just be `run(AddEmailToUser @data.email, current_user)`?", but...

Can we leave `TransferOmniauthInformation` so that later if we want to pull more info out we can?  It could be renamed `TransferOmniauthData` if that makes more sense w.r.t. other new names.